src/document.rs

```diff
@@ -264,17 +258,79 @@ impl Document {
                 buf
             };
         }
+        Ok(html.to_owned())
+    }
 
-        let options = LiquidOptions { file_system: Some(source.to_owned()), ..Default::default() };
+    /// Renders excerpt and adds it to attributes of the document.
+    pub fn render_excerpt(&mut self,
+                          context: &mut Context,
+                          source: &Path,
+                          default_excerpt_separator: &str)
+                          -> Result<String> {
+        let excerpt = {
+            let excerpt_separator: &str = self.attributes
+                .get("excerpt_separator")
+                .and_then(|attr| attr.as_str())
+                .unwrap_or(default_excerpt_separator);
+
+            if let Some(excerpt_str) = self.attributes
+                .get("excerpt")
+                .and_then(|attr| attr.as_str()) {
+                excerpt_str.to_string()
+            } else if excerpt_separator.is_empty() {
+                "".to_owned()
+            } else {
+                self.content.split(excerpt_separator)
+                    .filter(|chunk| !chunk.is_empty())
+                    .next() // first non-empty chunk
+                    .map(|s| s.to_string())
+                    .unwrap_or(self.content.to_string())
```

Comment:
It's not a huge thing but you can save on allocation and code if you make `excerpt` a `&str` instead of an owned String. You'd have to move it into a block for creating `excerpt_html` to avoid the dangling reference to `self.attributes`, though.